src/Procuration/ElectionContext.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+namespace AppBundle\Procuration;
+
+use AppBundle\Entity\Election;
+use Doctrine\Common\Collections\Collection;
+use Symfony\Component\Validator\Constraints as Assert;
+
+class ElectionContext implements \Serializable
+{
+    /**
+     * @Assert\Count(min=1, minMessage="procuration.election_context.min_count")
+     */
+    private $elections = [];
+
+    private $cachedElectionIds;
+
+    /**
+     * @return int[]|null
+     */
+    public function getIds(): ?array
```

Comment:
getCachedIds

or, you shoud check if cachedElectionIds is empty, then loop over elections